backend/server.js

```diff
@@ -24,55 +25,99 @@ function authenticateToken(req, res, next) {
 }
 
 // Create a new post
-app.post("/posts", authenticateToken, async (req, res) => {
-  const { title, content, imageUrls } = req.body;
+const upload = multer({ storage: multer.memoryStorage() });
 
-  try {
-    const newPost = await prisma.post.create({
-      data: {
-        title,
-        content,
-        updatedAt: null, // Ensure updatedAt is null when the post is created
-        user: { connect: { username: req.user.name } },
-        images: {
-          create: imageUrls.map((url) => ({ url })),
+app.post(
+  "/posts",
+  authenticateToken,
+  upload.array("images"),
+  async (req, res) => {
+    const { title, content } = req.body;
+    const images = req.files || []; // Provide a default empty array if req.files is undefined
```

Comment:
you can move this down into the try statement since if title is not provided, then it would just return 400 error

Reply:
Fixed! than you